### ISSUE TITLE
drop_chunks works even with default_transaction_read_only=on

### ISSUE THREAD

**Issue by tylerfontaine** [timescale/timescaledb#1912]
**Relevant system information:**
 - PostgreSQL 12.2 on x86_64-pc-linux-gnu, compiled by gcc, a 67f83fccf p a91f14b60b, 64-bit
 - TimescaleDB version (output of `\dx` in `psql`): 1.7.0
 - Installation method: [e.g., "using Docker", "apt install", "source"]: Yum

**Describe the bug**
calling drop_chunks() on a hypertable successfully drops chunks even with transactions set to read_only.
It's worth noting it fails as expected when `cascade_to_materializations` is set to true, and there are materializations to cascade to.

**To Reproduce**
1. create a hypertable and add some data.
2. Create a continuous agg on that hypertable
2. set transactions to read only `set default_transaction_read_only=on;`
3. Call drop_chunks:
```
SELECT count(*)
FROM drop_chunks(INTERVAL '4 days', 'some_hypertable', cascade_to_materializations => TRUE);
```
It will fail: 
```
ERROR:  cannot execute DELETE in a read-only transaction
CONTEXT:  SQL statement "DELETE FROM _timescaledb_internal._materialized_hypertable_5 AS D WHERE D.chunk_id = $1"
```
Set cascade_to_materializations to FALSE
```
SELECT count(*)
FROM drop_chunks(INTERVAL '4 days', 'some_hypertable', cascade_to_materializations => FALSE);
```
and it will succeed.

This ALSO succeeds regardless of the cascade setting if the hypertable does not have any materializations attached to it.

**Expected behavior**
I would expect read_only transactions to cause drop_chunks to fail.

**Actual behavior**
Chunks were actually dropped if no CASCADE action happened.

**Comment by k-rus** [comment]
@pmwkaa Have you fixed it?

**Comment by svenklemm** [comment]
This was fixed by #1997
